### PR TITLE
kickstart: init at 0.4.0

### PR DESCRIPTION
**Pull request by gaelreyrol**
## Description of changes

This PR packages the tool: https://github.com/Keats/kickstart

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by donovanglover** [comment]
Result of `nixpkgs-review pr 248160` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>kickstart</li>
  </ul>
</details>

**Review by donovanglover — APPROVED** [review]
LGTM

File: pkgs/by-name/ki/kickstart/package.nix [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    "--skip=generation::tests::can_generate_from_remote_repo_with_subdir"
+    "--skip=generation::tests::can_generate_from_remote_repo"
+  ];
+
+  passthru = {
+    updateScript = nix-update-script { };
+    tests.version = testers.testVersion {
+      inherit version;
+      package = kickstart;
+      command = "kickstart --version";
```
Comment (h7x4) [inline pkgs/by-name/ki/kickstart/package.nix:33]: This function already sources the version to test for from the package. Also `<mainProgram> --version` is the default command

https://github.com/NixOS/nixpkgs/blob/366b54b81097701b05276d1b9b63622c558abe17/pkgs/build-support/testers/default.nix#L59-L61

```suggestion
      package = kickstart;
```

**Review by h7x4 — APPROVED** [review]
Result of `nixpkgs-review pr 248160` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>kickstart</li>
  </ul>
</details>

Built and tested with `results/kickstart/bin/kickstart -o test-project https://github.com/Keats/rust-cli-template`, followed by mashing the enter button and inspecting the result.

Code is clean, tests pass, good stuff!

**Pull request merged**